x-pack/plugins/ml/common/types/data_frame_analytics.ts

```diff
@@ -61,63 +59,18 @@ interface GenericAnalysis {
   [key: string]: Record<string, any>;
 }
 
-export type AnalysisConfig =
-  | OutlierAnalysis
-  | RegressionAnalysis
-  | ClassificationAnalysis
-  | GenericAnalysis;
-
-export interface DataFrameAnalyticsConfig {
-  id: DataFrameAnalyticsId;
-  description?: string;
-  dest: {
-    index: IndexName;
-    results_field: string;
-  };
-  source: {
-    index: IndexName | IndexName[];
-    query?: estypes.QueryDslQueryContainer;
-    runtime_mappings?: RuntimeMappings;
-  };
-  analysis: AnalysisConfig;
-  analyzed_fields?: {
-    includes?: string[];
-    excludes?: string[];
-  };
-  model_memory_limit: string;
-  max_num_threads?: number;
-  create_time: number;
-  version: string;
-  allow_lazy_start?: boolean;
+export type AnalysisConfig = estypes.MlDataframeAnalysisContainer | GenericAnalysis;
```

Comment:
as far as i can tell, `GenericAnalysis` isn't needed.
It would be good if it can be removed and we can have predictable types

Reply:
Updated in f62dad344f36fd8fcd405c98394d91606b37be08